packages/daf-core/src/graphql/graphql-core.ts

```diff
@@ -113,52 +105,217 @@ function transformFindInput(input: FindInput): FindManyOptions {
       if (item.not === true) {
         where[item.column] = Not(where[item.column])
       }
+      return where
     }
-    result['where'] = where
   }
-  return result
+}
+
+function decorateQB(
+  qb: SelectQueryBuilder<any>,
+  tablename: string,
+  input: FindInput,
+): SelectQueryBuilder<any> {
+  if (input?.skip) qb = qb.skip(input.skip)
+  if (input?.take) qb = qb.take(input.take)
+
+  if (input?.order) {
+    for (const item of input.order) {
+      qb = qb.orderBy(`${escape(tablename)}.${escape(item.column)}`, item.direction)
+    }
+  }
+  return qb
+}
+
+async function checkAuthIdentity(did: string, authenticatedDid?: Promise<string>): Promise<boolean> {
+  if (!authenticatedDid) {
+    return true
+  }
+  const authenticatedDidString = await authenticatedDid
+  if (authenticatedDidString !== did) {
+    throw new Error('can not access restricted properties of another identity')
+  }
+  return true
 }
 
 export interface FindArgs {
   input?: FindInput
 }
 
 const messages = async (_: any, args: FindArgs, ctx: Context) => {
-  return (await ctx.agent.dbConnection).getRepository(Message).find(transformFindInput(args.input))
+  const where = createWhereObject(args.input)
+  let qb = (await ctx.agent.dbConnection)
+    .getRepository(Message)
+    .createQueryBuilder('message')
+    .leftJoinAndSelect('message.from', 'from')
+    .leftJoinAndSelect('message.to', 'to')
+    .where(where)
+  qb = decorateQB(qb, 'message', args.input)
+  if (ctx.authenticatedDid) {
+    const authIdent = await ctx.authenticatedDid
+    qb = qb.andWhere(
+      new Brackets(qb => {
+        qb.where('message.to = :ident', { ident: authIdent }).orWhere('message.from = :ident', {
+          ident: authIdent,
+        })
+      }),
+    )
+  }
```

Comment:
If we change `authenticatedDid` to `string` then this can be simplified like that:

```typescript
if (ctx.authenticatedDid) {
      qb = qb.andWhere(
       new Brackets(qb => {
          qb.where('message.to = :ident', { ident: ctx.authenticatedDid })
          .orWhere('message.from = :ident', { ident: ctx.authenticatedDid})
       }),
     )
   }
```

Reply:
https://github.com/apollographql/apollo-server/issues/225